File: solutions/0001.two_sum/Solution.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> records = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int mate = target - nums[i];
            if (null != records.get(mate)) {
                return new int[]{records.get(mate), i};
            }

            records.put(nums[i], i);
        }

        return null;
    }
}

File: solutions/0001.two_sum/solution.go
package solution

func twoSum(nums []int, target int) []int {
	records := make(map[int]int, len(nums))

	for i := 0; i < len(nums); i++ {
		mate := target - nums[i]
		if mateIdx, ok := records[mate]; ok {
			return []int{mateIdx, i}
		}

		records[nums[i]] = i
	}

	return nil
}
